### ISSUE TITLE
Notifications such as "could not access source file" appear on top of the keyboard

### ISSUE THREAD

**Issue by marcel263** [PhilippC/keepass2android#1937]
Notification bubbles should not be overlaid on the keyboard as it makes password antry difficult.

Version 1.09c 
Android 11 OneUI 3.1

**Comment by NumbGnat** [comment]
Are you able to get a screenshot @marcel263 ?

That might help illustrate the problem better.

**Comment by marcel263** [comment]
> Are you able to get a screenshot @marcel263 ?
> 
> That might help illustrate the problem better.

Dear,

the video included shows the problem.

Message should not appear on top of keyboard. It is like needless scrollbar or unskippable ads, very cumbersome. Obligatory messages should appear above the keyboard area.


https://user-images.githubusercontent.com/76437842/164773344-3c226d36-7d7f-46a9-88a4-cd583f050dda.mp4

**Comment by NumbGnat** [comment]
> the video included shows the problem.

Ahh... Yes... That helps. I thought perhaps that was the case. I opened Issue #1441 a while back, which is closely related.

@PhilippC didn't comment on it, so I was thinking perhaps he doesn't have any control over where the notice is displayed. That may be controlled by Android. 😕

**Comment by AvverbioPronome** [comment]
Yeah, this happens every time if your database is somewhere remote and it's not good at all.

**Comment by StrontRont** [comment]
It happens with WebDav file and Xperia 5 II with a 21:9 screen as well.

**Comment by jimmyjimjames** [comment]
Yeah, this issue drives me crazy.  I use a google drive file for my DB, and it keeps flashing a message up right in the middle of my keyboard as I'm trying to type my password.  Relocating the message to the top of the screen would likely solve this problem.

**Comment by Dricc123** [comment]
I love this app. This is a small thing. But when are they going to do something about this? the bug has been there for long now. or at least give us a statement saying it can't be fixed, so we stop asking.

**Comment by NumbGnat** [comment]
I'm pretty sure this is a standard Android toast message. Many of my other apps display their toast messages in the same spot.

So I did a little digging and landed on a [Codepath Guide for Displaying Toasts](https://guides.codepath.com/android/Displaying-Toasts)

There used to be a way for developers to specify the alignment of Toast messages, by calling `setGravity()`, but according to the [Android Toast Documentation](https://developer.android.com/reference/android/widget/Toast.html), as of Android 11 (September 2020), that was depreciated.  😕

So, I guess we're stuck with the `Toast()` messages in the location Android has chosen to force them into. 🙄

**Comment by Dricc123** [comment]
@NumbGnat what you're writing makes perfect sense because I didn't have the issue on my other phone that has an older version of Android installed. Now at least we know.

**Comment by d2718nis** [comment]
Unfortunately this is an issue on both Android 13 and 14. While I was researching this topic, I stumbled upon two possible solutions:
- The deprecation of `setGravity()` only applies to **text toasts** specifically. Unfortunately, the `setView()` is also deprecated, which means creating a custom Toast is not the most reliable way to solve it (Android developers are most likely to remove it completely in the future)
- Utilize the [Snackbar](https://developer.android.com/reference/com/google/android/material/snackbar/Snackbar). I believe there are some tools in [Xamarin CommunityToolkit](https://github.com/xamarin/XamarinCommunityToolkit/tree/main) that might help with that, although this project doesn't seem to have it among its dependencies

Both options won't allow showing toasts while the app is in the background, since Android R only the basic text toasts are allowed in such fashion. Not sure if this would matter for the current workflow.

Also, this is a related issue: https://github.com/PhilippC/keepass2android/issues/2069

**Comment by Wh1rledPeas** [comment]
> - Utilize the [Snackbar](https://developer.android.com/reference/com/google/android/material/snackbar/Snackbar). I believe there are some tools in [Xamarin CommunityToolkit](https://github.com/xamarin/XamarinCommunityToolkit/tree/main) that might help with that, although this project doesn't seem to have it among its dependencies

Looks like your research paid off! Thanks for digging into this for all of us!

Based on how many issues were resolved and closed with Phillips most recent update on this, it looks like you saved a ton of people a lot of headache!
